plugins/workspace-backpack/src/backpack.d.ts

```diff
@@ -0,0 +1,174 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/**
+ * @fileoverview A backpack that lives on top of the workspace.
+ * @author kozbial@google.com (Monica Kozbial)
+ */
+import * as Blockly from 'blockly/core';
+import { BackpackOptions } from './options';
+/**
+ * Class for backpack that can be used save blocks from the workspace for
+ * future use.
+ * @param {!Blockly.WorkspaceSvg} workspace The workspace to sit in.
+ * @implements {Blockly.IAutoHideable}
+ * @implements {Blockly.IPositionable}
+ * @extends {Blockly.DragTarget}
+ */
+export declare class Backpack extends Blockly.DragTarget {
+    /**
+     * Constructor for a backpack.
+     * @param {!Blockly.WorkspaceSvg} targetWorkspace The target workspace that
+     *     the backpack will be added to.
+     * @param {!BackpackOptions=} backpackOptions The backpack options to use.
+     */
+    constructor(targetWorkspace: Blockly.WorkspaceSvg, backpackOptions?: BackpackOptions);
+    /**
+     * Initializes the backpack.
+     */
+    init(): void;
+    /**
+     * Disposes of workspace search.
+     * Unlink from all DOM elements and remove all event listeners
+     * to prevent memory leaks.
+     */
+    dispose(): void;
+    /**
+     * Returns the backpack flyout.
+     * @return {?Blockly.IFlyout} The backpack flyout.
+     * @public
+     */
+    getFlyout(): Blockly.IFlyout | null;
+    /**
+     * Returns the bounding rectangle of the drag target area in pixel units
+     * relative to viewport.
+     * @return {?Blockly.utils.Rect} The component's bounding box. Null if drag
+     *   target area should be ignored.
+     */
+    getClientRect(): Blockly.utils.Rect | null;
+    /**
+     * Returns the bounding rectangle of the UI element in pixel units relative to
+     * the Blockly injection div.
+     * @return {!Blockly.utils.Rect} The component’s bounding box.
+     */
+    getBoundingRectangle(): Blockly.utils.Rect;
+    /**
+     * Positions the backpack.
+     * It is positioned in the opposite corner to the corner the
+     * categories/toolbox starts at.
+     * @param {!Blockly.MetricsManager.UiMetrics} metrics The workspace metrics.
+     * @param {!Array<!Blockly.utils.Rect>} savedPositions List of rectangles that
+     *     are already on the workspace.
+     */
+    position(metrics: Blockly.MetricsManager.UiMetrics, savedPositions: Array<Blockly.utils.Rect>): void;
```

Comment:
```suggestion
    position(metrics: Blockly.MetricsManager.UiMetrics, savedPositions: Blockly.utils.Rect[]): void;
```

Here and everywhere there's an array type.

Reply:
I fixed the TypeScript types from `Array<>` to `[]`, but left the JSDoc declarations as is. Let me know if I should be fixing JSDocs as well.